File: CPP/Hardrive.cpp
#include "Komponen.cpp"
#pragma once

class Harddrive : public Komponen {
    private:
        string tipeDrive;
        
    public:
        // Konstruktor
        Harddrive(string merk = "", string nama = "", string tipeDrive = "") 
            : Komponen(merk, nama) {
            this->tipeDrive = tipeDrive;
        }
        
        // Destruktor
        ~Harddrive() {
        }
        
        // Getter
        string getTipeDrive() const {
            return tipeDrive;
        }
        
        // Setter
        void setTipeDrive(string tipeDrive) {
            this->tipeDrive = tipeDrive;
        }
        
        // Method info
        string info() override {
            return "Harddrive: " + tipeDrive + " (Merk: " + merk + ", Nama: " + nama + ")";
        }
    };

File: Python/Hardrive.py
from Komponen import Komponen

class Harddrive(Komponen):
    def __init__(self, merk="", nama="", tipe_drive=""):
        super().__init__(merk, nama)
        self.tipe_drive = tipe_drive

    def get_tipe_drive(self):
        return self.tipe_drive

    def set_tipe_drive(self, tipe_drive):
        self.tipe_drive = tipe_drive

    def info(self):
        return f"Harddrive: {self.tipe_drive} (Merk: {self.get_merk()}, Nama: {self.get_nama()})"